ec/ecresource/projectresource/security.go

```diff
@@ -43,8 +47,77 @@ func NewSecurityProjectResource() *Resource[resource_security_project.SecurityPr
 
 type securityModelReader struct{}
 
+// productTypesOrderInsensitivePlanModifier ignores order differences in product_types list
+type productTypesOrderInsensitivePlanModifier struct{}
```

Comment:
Based on the comment here, I think we actually want to create a [custom type](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/types/custom) for the product types. That type can then implement [`ListValuableWithSemanticEquals`](https://github.com/hashicorp/terraform-plugin-framework/blob/7fb50058928e09bae9289d436c60b576270d1bf5/types/basetypes/list_value.go#L32) which:
* Applies the same logic as this plan modifier during the planning phase
* Critically, applies the same logic during the post application checks which will avoid the inconsistent state error I imagine we'd get if the TF config defined product types in a different order to what was returned by the API.